firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Util/vector.txt

```diff
@@ -0,0 +1,60 @@
+Scenario: The function in Util Vectors works correctly
+  GIVEN("The util resize function")
```

Comment:
The `resize` method should be tested with the following possible conditions:
1. We try to resize a `Vector` beyond its maximum allowed size. This should fail with an error code. Afterwards, the `Vector` should behave as if we never tried to do a resize at all. The data in the `Vector` before the resize should not be corrupted. 
2. We try to resize a `Vector` to a larger length than its current length. The data in the `Vector` before the resize should not be corrupted.
3. We try to resize a `Vector` to a smaller length than its current length. The `Vector` should behave as if the data after the new size no longer exists, i.e. elements at those indices get overwritten by `push_back`, while attempts to interact with those indices using `erase` would fail with bounds errors.

In every condition, after the resize:
- The Vector should correctly report its size (is it empty? is it full? how much capacity is available? what is its size?).
- All indices which we expect to be out-of-bounds (e.g. for erase) should actually be out-of-bounds, while all indices which we expect to be in-bounds (e.g. for erase) should actually be in-bounds.

Reply:
@raavilagoo It's not clear to me that this has been fully covered (e.g. the push_back tests in condition 3), but let's merge this PR as-is and any remaining improvements will be made in future PRs.